test/e2e/scheduling/priorities.go

```diff
@@ -0,0 +1,713 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		pods, err := framework.PodsCreatedByLabel(f.ClientSet, f.Namespace.Name, "scheduler-priority-selector-spreading", int32(len(nodeList.Items)), labels.SelectorFromSet(labels.Set(map[string]string{"statefulset": "ss1"})))
+		Expect(err).NotTo(HaveOccurred())
+		By("Ensuring each pod is running")
+
+		result := map[string]bool{}
+		for _, pod := range pods.Items {
+			if pod.DeletionTimestamp != nil {
+				continue
+			}
+			err = f.WaitForPodRunning(pod.Name)
+			Expect(err).NotTo(HaveOccurred())
+			result[pod.Spec.NodeName] = true
+		}
+		By("Verify the pods were scheduled to each node")
+		if len(nodeList.Items) != len(result) {
+			framework.Failf("Pods are not spread to each node")
+		}
+	})
+
+	It("Pod should be prefer scheduled to node that satisify the NodeAffinity", func() {
+		nodeName := GetNodeThatCanRunPod(f)
+		By("Trying to apply a label on the found node.")
+		k := fmt.Sprintf("kubernetes.io/e2e-%s", "node-topologyKey")
+		v := "topologyvalue"
+		framework.AddOrUpdateLabelOnNode(cs, nodeName, k, v)
+		framework.ExpectNodeHasLabel(cs, nodeName, k, v)
+		defer framework.RemoveLabelOffNode(cs, nodeName, k)
+
+		// make the nodes have balanced cpu,mem usage ratio
+		createBalancedPodForNodes(f, cs, ns, nodeList.Items, podRequestedResource, 0.6)
+		By("Trying to relaunch the pod, now with labels.")
+		labelPodName := "pod-with-node-affinity"
+		pod := createPausePod(f, pausePodConfig{
+			Name: labelPodName,
+			Affinity: &v1.Affinity{
+				NodeAffinity: &v1.NodeAffinity{
+					PreferredDuringSchedulingIgnoredDuringExecution: []v1.PreferredSchedulingTerm{
+						{
+							Preference: v1.NodeSelectorTerm{
+								MatchExpressions: []v1.NodeSelectorRequirement{
+									{
+										Key:      "kubernetes.io/hostname",
+										Operator: v1.NodeSelectorOpIn,
+										Values:   []string{nodeName},
+									}, {
+										Key:      k,
+										Operator: v1.NodeSelectorOpIn,
+										Values:   []string{v},
+									},
+								},
+							},
+							Weight: 20,
+						},
+					},
+				},
+			},
+		})
+		By("Wait the pod becomes running.")
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+		labelPod, err := cs.Core().Pods(ns).Get(labelPodName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		By("Verify the pod was scheduled to the expected node.")
+		Expect(labelPod.Spec.NodeName).To(Equal(nodeName))
+	})
+
+	It("Pod should be schedule to node that satisify the PodAffinity", func() {
+		By("Trying to launch a pod with a label to get a node which can launch it.")
+		pod := runPausePod(f, pausePodConfig{
+			Name:      "with-label-security-s1",
+			Labels:    map[string]string{"service": "S1"},
+			Resources: podRequestedResource,
+		})
+		nodeName := pod.Spec.NodeName
+
+		By("Trying to apply label on the found node.")
+		k := fmt.Sprintf("kubernetes.io/e2e-%s", "node-topologyKey")
+		v := "topologyvalue"
+		framework.AddOrUpdateLabelOnNode(cs, nodeName, k, v)
```

Comment:
Yes, prefer to add two nodes in this topology; current case is "similar" to node affinity.